File: DYObjectiveCApp/DYObjectiveCApp/ViewController.m
//
//  ViewController.m
//  DYObjectiveCApp
//
//  Created by hua on 2022/7/6.
//

#import "ViewController.h"
#import "PTWebViewController.h"
@import DingYue_iOS_SDK;

@interface ViewController ()<DYMPayWallActionDelegate>
@property (weak, nonatomic) IBOutlet UIButton *goPurchaseButton;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = UIColor.whiteColor;
}
- (IBAction)goPurchaseAction:(id)sender {
    [DYMobileSDK trackWithEvent:@"evenName" extra:nil user:nil];

    //使用DingYue内购页
    Subscription *defaultProduct = [[Subscription alloc] initWithType:@"SUBSCRIPTION" name:@"WEEK" platformProductId:@"testWeek" subscriptionDescription:@"test" period:@"week" price:@"7.99" currencyCode:@"USD" countryCode:@"US"];
    [DYMobileSDK showVisualPaywallWithProducts:@[defaultProduct] rootController:self extras:@{} completion:^(NSString * receipt, NSArray<NSDictionary<NSString *,id> *> * purchasedInfo, DYMError * error) {
            
    }];
}

#pragma mark - DYMPayWallActionDelegate
//实现隐私政策和使用协议action
-(void)clickTermsActionWithBaseViewController:(UIViewController *)baseViewController {
    PTWebViewController *vc = [[PTWebViewController alloc] init];
    vc.url = @"https://www.caretiveapp.com/tou/XXXXX/";
    vc.title = @"Terms of Service";
    UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController:vc];
    [baseViewController presentViewController:nav animated:YES completion:nil];
}
-(void)clickPrivacyActionWithBaseViewController:(UIViewController *)baseViewController {
    PTWebViewController *vc = [[PTWebViewController alloc] init];
    vc.url = @"https://www.caretiveapp.com/pp/XXXXX/";
    vc.title = @"Privacy Policy";
    UINavigationController *nav = [[UINavigationController alloc] initWithRootViewController:vc];
    [baseViewController presentViewController:nav animated:YES completion:nil];
}

- (void)clickCloseButtonWithBaseViewController:(UIViewController *)baseViewController {
    NSLog(@"点击了内购页上的关闭按钮");
}
- (void)payWallDidAppearWithBaseViewController:(UIViewController *)baseViewController {
    NSLog(@"订阅内购页显示");
}
- (void)payWallDidDisappearWithBaseViewController:(UIViewController *)baseViewController {
    NSLog(@"订阅内购页关闭");
}
@end

File: DYSwiftApp/DYSwiftApp/ViewController.swift
//
//  ViewController.swift
//  DYSwiftApp
//
//  Created by hua on 2022/7/6.
//

import UIKit
import DingYue_iOS_SDK

class ViewController: UIViewController{
    lazy var purchaseBtn: UIButton = {
        let btn = UIButton(type: .custom)
        btn.frame = CGRect(x: 0, y: 0, width: 200, height: 50)
        btn.center = self.view.center
        btn.setTitle("go Purchase", for: [])
        btn.setTitleColor(UIColor.black, for: [])
        btn.addTarget(self, action: #selector(goPurchase), for: .touchUpInside)
        return btn
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        self.view.backgroundColor = .white
        self.view.addSubview(purchaseBtn)
    }

    @objc func goPurchase(){
        let product1 = Subscription(type: "SUBSCRIPTION", name: "Week", platformProductId: "testWeek", price: "12.99", currencyCode: "USD", countryCode: "USD")
        let product2 = Subscription(type: "SUBSCRIPTION", name: "MONTH", platformProductId: "testMonth", price: "34.99", currencyCode: "USD", countryCode: "USD")
        
        DYMobileSDK.showVisualPaywall(products: [product1,product2], rootController: self) { receipt, purchaseResult, error in
            if error == nil {
                print("(DingYueSDK): 购买成功！")
            }
        }
    }
}
extension ViewController:DYMPayWallActionDelegate {
    public func clickTermsAction(baseViewController: UIViewController) {
        //do some customed thing
        let vc = LBWebViewController.init()
        vc.url = "https://www.caretiveapp.com/tou/XXXXXXX/"
        vc.title = "Terms_of_Service"
        let nav = UINavigationController.init(rootViewController: vc)
        nav.modalPresentationStyle = .fullScreen
        baseViewController.present(nav, animated: true)
    }

    public func clickPrivacyAction(baseViewController: UIViewController) {
        let vc = LBWebViewController.init()
        vc.url = "https://www.caretiveapp.com/pp/XXXXXXX/"
        vc.title = "Privacy_Policy"
        let nav = UINavigationController.init(rootViewController: vc)
        nav.modalPresentationStyle = .fullScreen
        baseViewController.present(nav, animated: true)
    }

    func clickCloseButton(baseViewController: UIViewController) {
        print("点击内购页上的关闭按钮")
    }
    func payWallDidAppear(baseViewController: UIViewController) {
        print("内购页显示")
    }
    func payWallDidDisappear(baseViewController: UIViewController) {
        print("内购页消失")
    }
}

